packages/tpetra/core/src/Tpetra_withLocalAccess.hpp

```diff
@@ -209,101 +253,327 @@ namespace Tpetra {
   //////////////////////////////////////////////////////////////////////
 
   /// \brief Declare that you want to access the given global object's
-  ///   local data in read-only mode, in the object's default memory
-  ///   space.
+  ///   local data in read-only mode.
   template<class GlobalObjectType>
   Details::LocalAccess<
     GlobalObjectType,
-    typename Details::DefaultMemorySpace<GlobalObjectType>::type,
-    Details::AccessMode::ReadOnly>
+    Details::read_only>
```

Comment:
We don't constrain the execution or memory spaces unless the user does, explicitly.  LocalAccess itself no longer "pre-constraints" them to their defaults.